notebooks/basics/xor.livemd

```diff
@@ -0,0 +1,117 @@
+# Modeling XOR with a neural network
+
+```elixir
+Mix.install([
+  {:axon, github: "elixir-nx/axon"},
+  {:nx, "~> 0.3.0", override: true},
+  {:exla, "~> 0.3.0"},
+  {:kino_vega_lite, "~> 0.1.3"}
+])
+
+Nx.Defn.default_options(compiler: EXLA)
+
+alias VegaLite, as: Vl
+```
+
+## Introduction
+
+In this notebook we try to create a model and learn it the **logical XOR**.
+
+Even though XOR seems like a trivial operation, it cannot be modeled using a single dense layer ([single-layer perceptron](https://en.wikipedia.org/wiki/Feedforward_neural_network#Single-layer_perceptron)). The underlying reason is that the classes in XOR are not linearly separable. We cannot draw a straight line to separate the points $(0,0)$, $(1,1)$ from the points $(0,1)$, $(1,0)$. To model this properly we need to turn to deep learning methods. They can separate data with much more complicated patterns within the data.
+
+## The model
+
+Let's start with the model. We need two inputs, since XOR has two operands, we then concatenate them into a single input vector with `Axon.concatenate/3`. Then we have one hidden layer and one output layer, both of them dense.
```

Comment:
```suggestion
Let's start with the model. We need two inputs, since XOR has two operands. We then concatenate them into a single input vector with `Axon.concatenate/3`. Then we have one hidden layer and one output layer, both of them dense.
```